trikot-viewmodels-declarative-flow/compose-flow/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/extensions/ImageResourceExtensions.kt

```diff
@@ -18,8 +24,53 @@ fun painterResource(imageResource: VMDImageResource?): Painter {
     val context = LocalContext.current
     imageResource?.let { resource ->
         resource.resourceId(context)?.let { id ->
-            return androidx.compose.ui.res.painterResource(id = id)
+            return if (painterResourceLegacyDrawableSupport) {
```

Comment:
This could be a different function and we would be able to pick the appropriate ones for optimal performance.

Reply:
I feel this should be a flag on the function rather than a global one.

Reply:
The thing is, `painterResource` is already called from a lot of places, among others in `VMDImage`..  we would need to pass this flag to alot of places.
Additionally, the need to support XML drawables depends on the VMDImageProvider mapping, _not_ on the context where the method is called.